File: app/models/posicao_fim_ano.rb
class PosicaoFimAno < ApplicationRecord
  belongs_to :ativo, foreign_key: :ativo_id

  validates :ano, numericality: true
  validates :preco_medio, numericality: true
  validates :posicao, numericality: true
  validates :custo_total, numericality: true

  # rails_admin do
  #   configure :preco_medio do
  #     pretty_value do
  #       valor.round(2)
  #     end
  #   end

  #   configure :custo_total do
  #     pretty_value do
  #       valor.round(2)
  #     end
  #   end
  # end
end

File: Stocks/Models/PosicaoFimAno.cs
namespace Stocks.Models;

public partial class PosicaoFimAno
{
    public int Id { get; set; }

    public int Ano { get; set; }

    public decimal PrecoMedio { get; set; }

    public decimal CustoTotal { get; set; }

    public int Posicao { get; set; }

    public int AtivoId { get; set; }

    public virtual Ativo Ativo { get; set; }
}
